tests/unit/adapter/test_maintenance_db.py

```diff
@@ -0,0 +1,56 @@
+# pylint: disable=unused-argument
+from pytest_mock import MockFixture
+from testcontainers.mongodb import MongoDbContainer
+
+from job_service.adapter import maintenance_db
+
+mongo = MongoDbContainer("mongo:5.0")
+mongo.start()
+DB_CLIENT = mongo.get_connection_client()
+
+
+def teardown_module():
+    mongo.stop()
+
+
+def setup_function():
+    DB_CLIENT.jobdb.drop_collection("maintenance")
+
+
+def test_set_and_get_status(mocker: MockFixture):
+    mocker.patch.object(
+        maintenance_db, "maintenance", DB_CLIENT.jobdb.maintenance
+    )
+
+    maintenance_db.set_status({"msg": "we upgrade chill", "pause": 1})
```

Comment:
Here we supply the set_status function with a dict, but in the API we send a MaintenanceRequest model